src/cursor/abstract_cursor.ts

```diff
@@ -457,9 +458,24 @@ export abstract class AbstractCursor<
    * cursor.rewind() can be used to reset the cursor.
    */
   async toArray(): Promise<TSchema[]> {
-    const array = [];
+    const array: TSchema[] = [];
+
+    // when each loop iteration ends,documents will be empty and a 'await (const document of this)' will run a getMore operation
     for await (const document of this) {
       array.push(document);
+      let docs = this.readBufferedDocuments();
+      if (this.transform != null) {
+        docs = await Promise.all(
+          docs.map(async doc => {
+            if (doc != null) {
+              return await this.transformDocument(doc);
+            } else {
+              throw Error;
```

Comment:
Not sure this is what you intended, this is throwing the `Error` constructor. Is it possible for readBufferedDocuments to return an array that has nullish values in it?

Reply:
Yep! Actually, since `readBufferedDocuments` removes non-nullish values, I changed it to a NonNullable<TSchema> (we'll see how EVG fares with this, but locally my tests are passing).